spec/data-types-and-values.html

```diff
@@ -170,38 +169,40 @@ <h1>
         </h1>
         <dl class="header"></dl>
         <emu-alg>
-          1. Let _valueString_ be ? ToString(_argument_.[[Value]]).
+          1. Let _value_ be ? Unbox(_argument_).
+          1. Let _valueString_ be ? ToString(_value_).
           1. Return the string-concatenation of *"Box("*, _valueString_, and *")"*.
         </emu-alg>
       </emu-clause>
 
-      <emu-clause id="sec-ecmascript-language-types-box-samevalue" type="abstract operation">
+      <emu-clause id="sec-ecmascript-language-types-box-unbox" type="abstract operation">
         <h1>
-          BoxSameValue (
-            _x_: a Box,
-            _y_: a Box,
+          Unbox (
+            _box_: a Box,
           )
         </h1>
-        <dl class="header"></dl>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It extracts the boxed Object from _box_, throwing if it has been boxed by a different Realm's Box constructor.</dd>
+        </dl>
         <emu-alg>
-          1. Let _xValue_ be _x_.[[Value]].
-          1. Let _yValue_ be _y_.[[Value]].
-          1. Return ! SameValue(_xValue_, _yValue_.)
+          1. Let _realm_ be the current Realm Record.
+          1. If _box_.[[Realm]] is not _realm_, throw a TypeError exception.

```

Comment:
We could potentially add a non-normative note that `_box_.[[Realm]]` is only accessed here and in `BoxSameValue`, and that since we only need its identity (and not its contents) engines are could just use an UUID so that boxes don't prevent realms to be garbage-collected.

Or is it too much into the "let's not write optimizations in the spec" territory?